docs/ext/commands/additional_info.rst

```diff
@@ -32,3 +32,29 @@ If your bot requires shard distribution across several machines, you should set
 This will prevent conflicts and race conditions. Discord API doesn't provide users with events related to application command updates,
 so it's impossible to keep the cache of multiple machines synced. Having only 1 machine with ``sync_commands`` set to ``True`` is enough
 because global registration of application commands doesn't depend on sharding.
+
+.. _why_params_and_injections_return_any:
+
+Why do ``Param`` and ``Injection``-related functions return ``Any``?
+--------------------------------------------------------------------
+
+If your editor of choice supports type-checking, you may have noticed that :func:`~ext.commands.Param`, :func:`~ext.commands.inject`,
+:func:`~ext.commands.injection` and :func:`~ext.commands.register_injection` do not have a specific return type, but at runtime these return :class:`~ext.commands.ParamInfo` and
+:class:`~ext.commands.Injection` respectively.
+
+A typical example of slash command might look like this: ::
+
+    @bot.slash_command(description="Replies with the given text!")
+    async def echo(
+        inter: disnake.ApplicationCommandInteraction,
+        text: str = commands.Param(description="Echo~")
+    ) -> None:
+        await inter.response.send_message(text)
+
+Here, you have two parameters in your command's function: ``inter``, an instance of :class:`disnake.ApplicationCommandInteraction`, and
+``text``, which is.. Noticed? You annotate ``text`` as ``str``, but at the same time, assign :class:`~ext.commands.ParamInfo` instance to it.
```

Comment:
```suggestion
``text``, which is somewhat unusual: you annotate ``text`` as ``str``, but at the same time, assign a :class:`~ext.commands.ParamInfo` instance to it.
```
maybe something like this instead?

Reply:
That was just a try to make documentation a bit less boring haha :P Yours look good too